packages/@aws-cdk/example-construct-library/lib/example-resource-utils.ts

```diff
@@ -0,0 +1,16 @@
+import * as cdk from '@aws-cdk/core';
+
+// This file contains utility functions used in the implementation of ExampleResource
```

Comment:
I kinda like a pattern where we put this into a subdirectory `private`, and NO files from the `private` directory should ever be exported.

Keeps it nice and obvious, I was thinking. We've inconsistently been applying this already in some places.

Okay with enshrining that into practice?

Reply:
This is not a good example of this pattern. The method `exampleResourceArnComponents` in this case should be in `example-resource.ts`.

I'm not convinced the `utils` suffix is a pattern we want to showcase for someone starting off with a construct library. None of the construct libraries I've dealt with so far (except stepfunctions) uses this pattern.

I would rather we drop this, and revisit it later?

Reply:
Obviously, this is a simplified version, as the functions are only used from one place. However, I've seen many PRs that add unnecessary public static methods to classes, so I think surfacing this pattern is valuable.

Would you prefer we change the name of the file, so it doesn't contain the `utils` fragment?

Reply:
> Would you prefer we change the name of the file, so it doesn't contain the utils fragment?

I can buy into that, yes

Reply:
Done: moved to `lib/private/`, and changed the name to `example-resource-common.ts`.